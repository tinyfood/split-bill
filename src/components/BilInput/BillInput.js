import React, { Component } from 'react'
import { connect } from 'react-redux'
import { editBill, deleteBill } from '../../actions/split'

class BillInput extends Component {
  state = {
    adding: false,
    keys: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '+', '0', 'delete']
  }

  handleKey = keyValue => {
    switch (keyValue) {
      case '+':
        break
      case 'delete':
        this.props.dispatch(deleteBill())
        break
      default:
        this.props.dispatch(editBill(keyValue))
        break
    }
  }

  render() {
    return (
      <div>
        {this.state.keys.map(key => (
          <div key={key} onClick={this.handleKey.bind(this, key)}>
            {key}
          </div>
        ))}
      </div>
    )
  }
}

export default connect()(BillInput)
