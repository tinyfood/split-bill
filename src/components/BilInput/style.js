import styled from 'styled-components'

export const Buttons = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  margin-bottom: 20px;
`

export const Button = styled.button`
  position: relative;
  width: 25%;
  margin: 12px;
  padding: 0;
  color: #4e4e4e;
  font-weight: 700;
  text-align: center;
  transition: all 0.1s linear;
  border: none;
  user-select: none;
  outline: none;
  cursor: pointer;

  &:active {
    transform: scale(1.2);
  }
`
