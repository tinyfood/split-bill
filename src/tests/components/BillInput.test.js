import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { BillInput } from '../../components/BilInput/BillInput'

test('should render BillInput correctly', () => {
  const wrapper = shallow(<BillInput />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
