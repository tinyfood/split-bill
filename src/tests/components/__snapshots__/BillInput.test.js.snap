// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BillInput correctly 1`] = `
<styled.div>
  <styled.button
    key="1"
    onClick={[Function]}
    type="button"
    value="1"
  >
    1
  </styled.button>
  <styled.button
    key="2"
    onClick={[Function]}
    type="button"
    value="2"
  >
    2
  </styled.button>
  <styled.button
    key="3"
    onClick={[Function]}
    type="button"
    value="3"
  >
    3
  </styled.button>
  <styled.button
    key="4"
    onClick={[Function]}
    type="button"
    value="4"
  >
    4
  </styled.button>
  <styled.button
    key="5"
    onClick={[Function]}
    type="button"
    value="5"
  >
    5
  </styled.button>
  <styled.button
    key="6"
    onClick={[Function]}
    type="button"
    value="6"
  >
    6
  </styled.button>
  <styled.button
    key="7"
    onClick={[Function]}
    type="button"
    value="7"
  >
    7
  </styled.button>
  <styled.button
    key="8"
    onClick={[Function]}
    type="button"
    value="8"
  >
    8
  </styled.button>
  <styled.button
    key="9"
    onClick={[Function]}
    type="button"
    value="9"
  >
    9
  </styled.button>
  <styled.button
    key="AC"
    onClick={[Function]}
    type="button"
    value="AC"
  >
    AC
  </styled.button>
  <styled.button
    key="0"
    onClick={[Function]}
    type="button"
    value="0"
  >
    0
  </styled.button>
  <styled.button
    key="delete"
    onClick={[Function]}
    type="button"
    value="delete"
  >
    <DeleteIcon />
  </styled.button>
</styled.div>
`;
